scripts/createOrReplaceDatabase.sh

```diff
@@ -59,24 +56,28 @@ while read line; do
 done < $configFilepath
 
 ## Improperly formatted or missing config file
-if [ -z "$username" ] || [ -z "$password" ]
+if [ -z "$username" ]
 then
-	echo "Could not find username and/or password in $configFilepath";
+	echo "No username in $configFilepath";
 	exit 1
 fi
 
-echo "Updating $dbName with $dbFilepath";
-echo "Backing up $dbName";
-
 ## Take archive of DB, drop it and create a new, empty one
 dbArchiveFile="$dbName.backup.dump"
-mysqldump -u$username -p$password -h$host -P$port $dbName > $dbArchiveFile
-echo "Compressing $dbArchiveFile";
-eval "gzip -f $dbArchiveFile";
-echo "Finished backing up $dbName";
+if ! mysql -u$username -h$host -P$port -e "use $dbName" 2> /dev/null;
+then
+	echo "Creating $dbName";
+	mysql -u$username -h$host -P$port -e "create database $dbName"
```

Comment:
Is this branch necessary since the db will get created on line 80?

Reply:
It was throwing a MySQL error when it was trying to a backup of a non-existent DB, so I had to put that branch in

Reply:
Could the logic just be rewritten as 
```
if $databaseDOESExists then
  # do backup stuff...
fi
```

Reply:
Right, but you could reverse the condition for controlling when you take a backup and remove this branch for creating a db since that happens later on line 80.

Reply:
I swapped the logic to remove the `!` part. I think it looks more readable now